Update the content of the `Using Object.keys() on primitives` section (#29981)
mdn/content: files/en-us/web/javascript/reference/global_objects/object/keys/index.md

### Using Object.keys() on primitives

> Non-object arguments are [coerced to objects](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object#object_coercion). Only strings may have own enumerable properties, while all other primitives(except [`undefined`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) and [`null`](/en-US/docs/Web/JavaScript/Reference/Operators/null), which will throw a {{jsxref("TypeError")}}) return an empty array.

Reviewer: ```suggestion
Non-object arguments are [coerced to objects](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object#object_coercion). [`undefined`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) and [`null`](/en-US/docs/Web/JavaScript/Reference/Operators/null) cannot be coerced to objects and throw a {{jsxref("TypeError")}} upfront. Only strings may have own enumerable properties, while all other primitives return an empty array.
```

```js
// Strings have indices as enumerable own properties
console.log(Object.keys("foo")); // ['0', '1', '2']

// Other primitives except undefined and null have no own properties
console.log(Object.keys(100)); // []
```

> ```js example-bad
> console.log(Object.keys(undefined)); // TypeError: Cannot convert undefined or null to object
> console.log(Object.keys(null)); // TypeError: Cannot convert undefined or null to object
> ```
> 

Reviewer: ```suggestion
```
